plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7902,10 +7903,13 @@ private List<TypeCoercionTestSetup> typeCoercionOnCreateTableAsSelectData()
                 .add(new TypeCoercionTestSetup("CHAR 'A '", "varchar", "'A '"))
                 .add(new TypeCoercionTestSetup("CHAR ' A'", "varchar", "' A'"))
                 .add(new TypeCoercionTestSetup("CHAR 'ABc'", "varchar", "'ABc'"))
+                .add(new TypeCoercionTestSetup("ARRAY[CHAR 'A']", "array(varchar)", "ARRAY['A']"))
```

Comment:
Can we add test for nested structural types ?

Reply:
Is it possible to add them for delta lake ?

Reply:
The current helper class of TestDeltaLakeConnectorTest can't those types nicely. Let me handle in follow-up with `TestSetup` refactoring.